app/controllers/ViewSubmissionsController.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.time.Instant
+
+import connectors.CustomsDeclareExportsMovementsConnector
+import controllers.actions.AuthenticatedAction
+import javax.inject.{Inject, Singleton}
+import models.notifications.NotificationFrontendModel
+import models.submissions.SubmissionFrontendModel
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.view_submissions
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ViewSubmissionsController @Inject()(
+  authenticate: AuthenticatedAction,
+  connector: CustomsDeclareExportsMovementsConnector,
+  mcc: MessagesControllerComponents,
+  viewSubmissionsPage: view_submissions
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage: Action[AnyContent] = authenticate.async { implicit request =>
+    val providerId = request.pid
+
+    for {
+      submissions <- connector.fetchAllSubmissions(providerId)
+      notifications <- Future.sequence(submissions.map(submission => connector.fetchNotifications(submission.conversationId, providerId)))
```

Comment:
This seems inefficient

For many submissions we are making many back end calls.

It'd be better just to query the back end `/notifications` with `providerId`/`pid` and then group by actionID

Reply:
I agree, it would be better. But this would require new endpoint on the BE. So I think this could be done as a separate tech debt ticket.
Also, we need to consider if we are able to handle that huge payload with all notifications in a single call.

Reply:
Fair point I'll raise a ticket